crates/findex/src/pyo3/py_api.rs

```diff
@@ -463,24 +160,21 @@ impl InternalFindex {
     #[allow(clippy::too_many_arguments)]
     pub fn search_wrapper(
         &mut self,
-        keywords: Vec<&str>,
+        keywords: Vec<ToKeyword>,
         master_key: &MasterKeyPy,
         label: &LabelPy,
         max_result_per_keyword: usize,
         max_depth: usize,
         fetch_chains_batch_size: usize,
         progress_callback: Option<PyObject>,
-        py: Python,
-    ) -> PyResult<HashMap<String, Vec<PyObject>>> {
+    ) -> PyResult<HashMap<KeywordPy, Vec<LocationPy>>> {
         self.progress_callback = match progress_callback {
             Some(callback) => callback,
             None => self.default_progress_callback.clone(),
         };
 
-        let keywords_set: HashSet<Keyword> = keywords
-            .iter()
-            .map(|keyword| Keyword::from(*keyword))
-            .collect();
+        let keywords_set: HashSet<Keyword> =
+            keywords.iter().map(|keyword| keyword.0.clone()).collect();
```

Comment:
Needless copy (you already have the ownership of the keywords):
```suggestion
        let keywords_set: HashSet<Keyword> =
            keywords.into_iter().map(|keyword| keyword.0).collect();
```